src/ui/viewmodels/TriggerConditionViewModel.cpp

```diff
@@ -513,7 +513,9 @@ void TriggerConditionViewModel::OnValueChanged(const IntModelProperty::ChangeArg
     }
     else if (args.Property == OperatorProperty)
     {
+        const auto nCurrentType = ra::itoe<TriggerOperandType>(GetValue(TargetTypeProperty));
         SetValue(HasTargetProperty, ra::itoe<TriggerOperatorType>(args.tNewValue) != TriggerOperatorType::None);
+        SetValue(HasTargetValueProperty, !IsParameterlessType(nCurrentType) && GetValue(HasTargetProperty));
```

Comment:
The `HasTargetSize` visibility flag is being updated when the `HasTargetProperty` changed. This logic should probably be moved there too:
```
    if (args.Property == HasTargetProperty)
    {
        const auto nTargetType = GetTargetType();
        SetValue(HasTargetSizeProperty, args.tNewValue && IsAddressType(nTargetType));
        SetValue(HasTargetValueProperty, args.tNewValue && !IsParameterlessType(nTargetType));
    }
```

Reply:
Yes, that's a better location.